packages/react-components/src/shared/src/mergeProps.ts

```diff
@@ -115,7 +116,6 @@ function merge(props: Record<string, any>, newProps: Record<string, any>) {
 
 // Taken from: https://stackoverflow.com/questions/51603250/typescript-3-parameter-list-intersection-type/51604379#51604379.
 type TupleTypes<T> = { [P in keyof T]: T[P] } extends { [key: number]: infer V } ? V : never;
```

Comment:
Would love to see a few tests for this type, it would also make it easier to understand for newcomes (and well established devs like me who keep forgetting how it works haha)